packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1515,7 +1515,7 @@ class Navigator extends StatefulWidget {
   final List<Page<dynamic>> pages;
 
   /// Called when [pop] is invoked but the current [Route] corresponds to a
-  /// [Page] found in the [pages] list.
+  /// [Page].
```

Comment:
It feels like we should check the page is in the pages list in https://github.com/flutter/flutter/blob/99f19d05c331a04054535cad1c8087916eeacd60/packages/flutter/lib/src/widgets/navigator.dart#L5073

Is there any use case this can happen?

Reply:
How about we make the docs agree with the code right now, in this PR?

Reply:
In the real world scenario, the page should be in the pages list. If it is not, it is either a bug in our code or a user error. It may not be ideal that we relax the language in the documentation.

Reply:
Hi @mleonhard I thought change the code would be as easy as adding
```dart
assert(widget.pages.contains(entry.route.settings as Page<dynamic>));
```
before this line https://github.com/flutter/flutter/blob/99f19d05c331a04054535cad1c8087916eeacd60/packages/flutter/lib/src/widgets/navigator.dart#L5074

or do I misunderstand the requirement?

Reply:
`Navigator.pop()` calls `onPopPage` even when the top-most widget is not in the `pages` list.  Adding an assert cannot change that behavior.  We must add a branch.  Adding a test is good, too.

Reply:
Ah I think I know what is going on. I tried to reproduce the original issue, and the main problem is this widget
```
CupertinoDialogAction(
          child: Text('Error Detail'),
          onPressed: () => Navigator.of(ctx)
              .push(errorDetailPage(text).createRoute(ctx)),
        ),
```

This tries to use imperative push to push a pagebased route. When I run the code, I immediately sees the following error
```
══╡ EXCEPTION CAUGHT BY WIDGET LIBRARY ╞════════════════════════════════════════════════════════════
The following assertion was thrown:
A page-based route should not be added using the imperative api. Provide a new list with the
corresponding Page to Navigator.pages instead.

When the exception was thrown, this was the stack:
#0      NavigatorState._debugCheckIsPagelessRoute.<anonymous closure> (package:flutter/src/widgets/navigator.dart:4517:31)
#1      NavigatorState._debugCheckIsPagelessRoute (package:flutter/src/widgets/navigator.dart:4522:6)
#2      NavigatorState.push (package:flutter/src/widgets/navigator.dart:4502:12)
#3      LoaderState._showErrorDialog.<anonymous closure>.<anonymous closure> (package:test2/main.dart:95:16)
#4      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:195:24)
#5      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:608:11)
#6      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:296:5)
#7      BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:267:7)
#8      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:157:27)
#9      GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:444:20)
#10     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:420:22)
#11     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:287:11)
#12     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#13     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#14     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
#15     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
#19     _invoke1 (dart:ui/hooks.dart:182:10)
#20     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:282:7)
#21     _dispatchPointerDataPacket (dart:ui/hooks.dart:96:31)
(elided 3 frames from dart:async)
════════════════════════════════════════════════════════════════════════════════════════════════════
E/AccessibilityBridge( 9631): Semantics node id does not equal ROOT_NODE_ID.
E/AccessibilityBridge( 9631): Semantics node id does not equal ROOT_NODE_ID.

```

This should already warn the developers that this is not allow. and the onpoppage crash is a byproduct of this error. 

Based on your error log, you didn't see the error message. I assume you are running on the release mode?

The correct fix is either make contactSupportPage a pageless route, or pass it into the Navigator.pages instead using the Navigator.push